### PR TITLE
Update comments for Socket.UseOnlyOverlappedIO

### LINKED ISSUE #34458: Missing Socket.UseOnlyOverlappedIO functionality?
This comment in the UseOnlyOverlappedIO setter suggests it's a nop because we don't support DuplicateAndClose:
https://github.com/dotnet/runtime/blob/15ec69e47b4dc56098e6058a11ccb6ae4d5d4fa1/src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs#L325-L329

But DuplicateAndClose was implemented here:
https://github.com/dotnet/runtime/pull/1858

Either the setter needs to be implemented, or the comment needs to be updated.

### PR DESCRIPTION
**Pull request by antonfirsov**
Closes #34458.

/cc @stephentoub

### CODE REVIEW

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs [hunk 1]
```diff
@@ -417,22 +417,15 @@ public bool Blocking
...
-            {
-                //
-                // This implementation does not support non-IOCP-based async I/O on Windows, and this concept is
-                // not even meaningful on other platforms.  This option is really only functionally meaningful
-                // if the user calls DuplicateAndClose.  Since we also don't support DuplicateAndClose,
-                // we can safely ignore the caller's choice here, rather than breaking compat further with something
-                // like PlatformNotSupportedException.
-                //
-            }
+            get { return false; }
```
Comment (stephentoub) [inline src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs:427]: Do we have any tests that validate it defaults to false and that the setter is a nop?

Comment (antonfirsov) [reply src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs:427]: Yes:
https://github.com/dotnet/runtime/blob/1edc35f8517bbb2fae8c0152de14bd105f9bac06/src/libraries/System.Net.Sockets/tests/FunctionalTests/SocketDuplicationTests.cs#L30-L38

**Review by karelz — APPROVED** [review]

**Review by geoffkizer — APPROVED** [review]

**Comment by antonfirsov** [comment]
Closing and reopening in an attempt to wake up the GitHub checks.

File: src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs [hunk 2]
```diff
@@ -417,22 +417,15 @@ public bool Blocking
             }
         }
 
+        // On .NET Framework, this property functions as a socket-level switch between IOCP-based and Win32 event based async IO.
+        // On that platform, setting UseOnlyOverlappedIO = true prevents assigning a completion port to the socket,
+        // allowing calls to DuplicateAndClose() even after performing asynchronous IO.
+        // .NET (Core) Windows sockets are entirely IOCP-based, and the concept of "overlapped IO" 
+        // does not exist on other platforms, therefore UseOnlyOverlappedIO is a dummy, compat-only property.
```
Comment (stephentoub) [inline src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs:424]:
```suggestion
        // On .NET Framework, this property functions as a socket-level switch between IOCP-based and Win32 event based async IO.
        // On that platform, setting UseOnlyOverlappedIO = true prevents assigning a completion port to the socket,
        // allowing calls to DuplicateAndClose() even after performing asynchronous IO.
        // .NET (Core) Windows sockets are entirely IOCP-based, and the concept of "overlapped IO"
        // does not exist on other platforms, therefore UseOnlyOverlappedIO is a dummy, compat-only property.
```

**Pull request merged**